### ISSUE TITLE
iffy diagnostics for struct defined in a const generic default that uses generics from outer struct

### ISSUE THREAD

**Issue by BoxyUwU** [rust-lang/rust#86651]
Given the following code: [link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=48fa3a6ccbc1e086e2377fc5abc7d36c)

```rust
#![feature(const_generics, const_generics_defaults)]
#![allow(incomplete_features)]

struct Foo<const N: usize, const M: usize = { struct Bar<const L: usize = N>; N }>;
```

The current output is:

```
error[E0401]: can't use generic parameters from outer function
 --> src/lib.rs:4:75
  |
4 | struct Foo<const N: usize, const M: usize = { struct Bar<const L: usize = N>; N }>;
  |                  - const parameter from outer function                    ^ use of generic parameter from outer function

```

Ideally the output should shouldn't talk about functions but struct definitions instead

**Comment by fmease** [comment]
Current output:

```
error[E0401]: can't use generic parameters from outer item
 --> src/lib.rs:4:75
  |
4 | struct Foo<const N: usize, const M: usize = { struct Bar<const L: usize = N>; N }>;
  |                  -                                       -                ^ use of generic parameter from outer item
  |                  |                                       |
  |                  const parameter from outer item         help: try introducing a local generic parameter here: `N,`
```

for

```rs
#![feature(generic_const_exprs)]
#![allow(incomplete_features)]

struct Foo<const N: usize, const M: usize = { struct Bar<const L: usize = N>; N }>;
```

**Comment by fmease** [comment]
I've fixed that a while ago namely in #115744. Closing as fixed.
